File: Behavioral/Lab20/C#/RealtorSubsystem.cs
using System;
using System.Collections.Generic;
using System.Text;
using C_.Entities;

namespace C_
{
    public class RealtorSubsystem : AbstractSubsystem
    {

        public override void MainAction(IEnumerable<AgencyObject> agencyObjects, bool notify = true)
        {
            var realtors = agencyObjects as IEnumerable<Realtor>;
            var sb = new StringBuilder();
            sb.AppendLine("Realtors portfolio:");
            foreach (var realtor in realtors)
            {
                sb.Append($"Realtor {realtor.NameAndSurname} has {realtor.RealEstates.Count} estates\n");
            }
            if (notify)
                Agency.GetInstance().Notify("Estates protfolio");
            Console.WriteLine(sb.ToString());
        }
        

    }
}

File: Behavioral/Lab19/Typescript/src/realtor_subsystem.ts
import { AbstractSubsystem } from "./abstract_subsystem";
import { Agency } from "./Entities/agency";
import { AgencyObject } from "./Entities/agency_object";
import { Realtor } from "./Entities/realtor"

export class RealtorSubsystem implements AbstractSubsystem{
    public MainAction(objects:Array<AgencyObject>, notify:boolean = true):void{
        const sb: string[] = [];
        const realtors = objects as Array<Realtor>;
        for (const realtor of realtors) {
            sb.push(`Realtor ${realtor.nameAndSurname} has ${realtor.RealEstates.length} estates\n`);
        }
        if(notify)
            Agency.Notify("estate");
        console.log(sb.join(""));
    }
}